addons/point_of_sale/static/src/app/screens/payment_screen/payment_screen.js

```diff
@@ -38,7 +38,17 @@ export class PaymentScreen extends Component {
         this.printer = useService("printer");
         this.payment_methods_from_config = this.pos.config.payment_method_ids
             .slice()
-            .sort((a, b) => a.sequence - b.sequence);
+            .sort((a, b) => a.sequence - b.sequence)
+            .filter((pm) => {
+                if (
+                    (pm.terminal_id && !pm.pos_payment_provider_id) ||
+                    (pm.pos_payment_provider_id && pm.pos_payment_provider_id.state === "disabled")
+                ) {
+                    return false;
+                } else {
+                    return true;
+                }
+            });
```

Comment:
how about not loading the payment method to the client app when the linked provider is disabled? That way, you don't need to filter it here in the client app.

Reply:
To do this, go to the `pos.payment.method` model and modify the "loading" domain.